modules/dkan/dkan_datastore/dkan_datastore.module

```diff
@@ -225,12 +204,9 @@ function dkan_add_resource($node) {
 function dkan_datastore_node_insert($node) {
   $type = dkan_datastore_node_type();
   if ($node->type == $type) {
-    if (!isset($node->feeds)) {
-      // Feeds only saves if there is a form present. We want this to work
-      // programmatically as well.
-      feeds_node_prepare($node);
-      feeds_node_update($node);
-    }
+    $resource = Resource::createFromDrupalNode($node);
```

Comment:
I wonder if it is an issue that this creates a manager state object in the DB for _every_ resource, even though it will only ever be used with datastore-compatable resources. I suppose maybe it doesn't hurt anything to have them all in there?

Reply:
mmm... doesn't the createFromDrupalNode() method checks some level of compatibility?